version: '3'

services:
  skygear-db:
    image: mdillon/postgis:9.5
    ports:
    - '5433:5432'
    volumes:
    - skygear-auth-data:/var/lib/postgresql/data

  skygear-gateway:
    image: quay.io/skygeario/skygear-gateway:git-ef8ac87
    depends_on:
    - 'skygear-db'
    ports:
    - '3000:3000'
    environment:
      HOST: '0.0.0.0:3000'
      STANDALONE: 'true'
      AUTH_LIVE_URL: 'http://skygear-auth:3001'
      STANDALONE_TENANT_CONFIG_FILE: '/standalone-tenant-config.yaml'
    volumes:
    - './standalone-tenant-config.yaml:/standalone-tenant-config.yaml'

  skygear-auth:
    image: quay.io/skygeario/skygear-auth:git-ef8ac87
    depends_on:
    - 'skygear-db'
    environment:
      HOST: '0.0.0.0:3001'
      STANDALONE: 'true'
      STANDALONE_TENANT_CONFIG_FILE: '/standalone-tenant-config.yaml'
    volumes:
    - './standalone-tenant-config.yaml:/standalone-tenant-config.yaml'

  myservice:
    build:
      context: ./myservice

volumes:
  skygear-auth-data: {}
